### PR TITLE
Upgraded React-Router to 2.8.1 and React to 15.6.1

### PR DESCRIPTION
**Pull request by JamesFTW**
Freight now has the same React-Router as Sentry, and a slightly newer version of React and ReactDOM.

Some rewiring was needed. 

Also edited webpack.config to cache static assests.
screen shot 2017-06-26 at 1 39 29 pm

### CODE REVIEW

File: static/components/TaskDetails.jsx [hunk 1]
```diff
@@ -137,14 +137,24 @@ var TaskDetails = React.createClass({
       div.className  = 'line';
       time.className = 'time';
 
+=======
```
Comment (benvinegar) [inline static/components/TaskDetails.jsx]: ⬆️

File: static/routes.jsx [hunk 2]
```diff
@@ -9,19 +9,25 @@ import Overview from "./components/Overview";
...
+  return (
+    <Route exact path="/" component={Layout}>
+      <IndexRoute component={Overview} />
+      <Route path="/deploy" component={CreateDeploy} />
+      <Route path="/tasks/:app/:env/:number" component={TaskDetails} />
+      <Route path="/deploys/:app/:env/:number" component={TaskDetails} />
+      <Route path="/:app/settings" component={AppSettings} />
+      <Route path="/:app/:env/:number" component={TaskDetails} />
+      <Route path="/:app" component={AppDetails} />
+      <Route
```
Comment (evralston) [inline static/routes.jsx]: weird wrapping on this last one

File: static/components/TaskSummary.jsx [hunk 3]
```diff
@@ -77,11 +77,11 @@ var TaskSummary = React.createClass({
     if (task.status === 'failed') {
       className += ' failed';
     } else if (task.status === 'cancelled') {
-      className += ' cancelled';
+      className += 'cancelled';
```
Comment (evralston) [inline static/components/TaskSummary.jsx]: if you're pulling the leading space out of here should you also take it out of ` failed`?  or does leaving it on `failed` mean this one shouldn't be removed?

File: static/components/TaskDetails.jsx [hunk 4]
```diff
@@ -104,8 +104,8 @@ var TaskDetails = React.createClass({
   },
 
   getPollingUrl() {
-    var params = this.getParams();
-    return '/deploys/' + params.app + '/' + params.env + '/' + params.number + '/';
+    var {app, env, number} = this.props.params;
+    return '/deploys/' + app + '/' + env + '/' + number + '/';
```
Comment (kjlundsgaard) [inline static/components/TaskDetails.jsx]: You could probably use template strings here.
```return `/deploys/${app}/${env}/${number}/`;```

Comment (JamesFTW) [inline static/components/TaskDetails.jsx]: Done.

**Comment by benvinegar** [comment]
Remove the chunk hash stuff – don't attempt to do this `url_for` stuff in
the same PR, please.

**Pull request merged**